interfaces/sourcegen/sourcegen/_HeaderFileParser.py

```diff
@@ -8,20 +8,53 @@
 import re
 from typing import List
 
-from ._dataclasses import HeaderFile, Func
-
+from ._dataclasses import HeaderFile, Func, Recipe
+from ._helpers import read_config
 
 _logger = logging.getLogger()
 
-_clib_path = Path(__file__).parent.joinpath("../../../include/cantera/clib").resolve()
+_clib_path = Path(__file__).parents[3] / "include" / "cantera" / "clib"
 _clib_ignore = ["clib_defs.h", "ctmatlab.h"]
 
+_data_path = Path(__file__).parent / "_data"
+
 class HeaderFileParser:
 
     def __init__(self, path: Path, ignore_funcs: List[str] = None):
         self._path = path
         self._ignore_funcs = ignore_funcs
 
+    @classmethod
+    def from_yaml(cls, ignore_files, ignore_funcs) -> List[HeaderFile]:
```

Comment:
Sure thing! I've been switching to `list` for the most part, but some of this code is backported from an old branch. :+1: on the suggestion of the method name ... since all of this is experimental, I switched both names.